<!--
Configurable launch file to start a Net USB camera.

You can uncomment the viewer at bottom if you want to view the image stream.
-->

<launch>

  <!-- Namespace to publish in -->
  <arg name="camera" default="camera" />

  <!-- You can select the camera to open by its GUID or IP address. In some cases
       if the camera is connected through a switch (not visible to ListCameras),
       it may only be possible to open a camera by its fixed IP address.
       
       For help configuring your camera or determining its IP address, see
       http://www.ros.org/wiki/prosilica_camera/Tutorials.
       
       By default, when neither argument is set, open the first camera found.
       If both are set, open by IP address and verify the GUID matches.
  -->
  <arg name="camera_id" default="0" />
  <arg name="color_mode" default="rgb24"/>


  <!-- Trigger mode: 'streaming', 'software' or 'hardware' -->
  <arg name="trigger_mode" default="streaming" />

  <!-- tf frame id associated with this camera -->
  <arg name="frame_id" default="$(arg camera)_optical_frame" />

  <group ns="$(arg camera)">

  <!-- The camera node -->
  <node name="$(arg camera)_driver" pkg="netusb_camera_driver" type="netusb_node" output="screen">
    <remap from="camera" to="$(arg camera)" />

    <param name="color_mode" value="$(arg color_mode)" />

    <param name="camera_id" value="$(arg camera_id)" />

    <param name="trigger_mode" value="$(arg trigger_mode)" />
    <param name="frame_id" value="$(arg frame_id)" />
  </node>

  <!-- If running in polled mode, change ns to match your service call argument. -->
  

      <!--
      Uncomment below if you want to view the image stream on a monochrome
      camera. (You will see the Bayer coded image on a color camera)

      You will have to run the following command first:
      $ rosmake image_view
      -->

    <!--
    <node pkg="image_view" type="image_view" respawn="false" name="prosilica_viewer_mono">
      <param name="autosize" type="bool" value="False" />
      <param name="window_name" type="str" value="'Prosilica Camera (Mono)'" />
      <remap from="image" to="image_raw" />
    </node>
    -->

      <!--
      Uncomment below if you want to view the image stream on a color camera.

      You will have to run the following command first:
      $ rosmake image_view image_proc
      -->

    <!--
    <node pkg="image_proc" type="image_proc" respawn="false" name="prosilica_image_proc" />
    <node pkg="image_view" type="image_view" respawn="false" name="prosilica_viewer_color">
      <param name="autosize" type="bool" value="False" />
      <param name="window_name" type="str" value="'Prosilica Camera (Color)'" />
      <remap from="image" to="image_color" />
    </node>
    -->

  </group>

</launch>
